students/DrewSmith/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,133 @@
+    database_name = database_name or config.DATABASE_NAME
+    directory = pathlib.Path(directory_name)
+
+    errors = [0, 0, 0]
+    counts = [0, 0, 0]
+    with connection:
+        database = connection.connection.get_database(name=database_name)
+
+        with open(directory / product_file, 'r') as file:
+            csv_file = csv.DictReader(file, fieldnames=("product_id",
+                                                        "description",
+                                                        "product_type",
+                                                        "quantity_available"))
+            records = list()
+            for index, record in enumerate(csv_file):
+                if index == 0:
+                    continue
+                try:
+                    record["quantity_available"] = int(record["quantity_available"])
+                except ValueError:
+                    errors[0] += 1
+                else:
+                    records.append(record)
+            counts[0] = len(records)
+            database["products"].insert_many(records)
+
+        with open(directory / customer_file, 'r') as file:
+            csv_file = csv.DictReader(file, fieldnames=("user_id",
+                                                        "name",
+                                                        "address",
+                                                        "zip_code",
+                                                        "phone_number",
+                                                        "email"))
+            records = [record for index, record in enumerate(csv_file) if index > 0]
+            counts[1] = len(records)
+            database["customers"].insert_many(records)
+
+        with open(directory / rentals_file, 'r') as file:
+            csv_file = csv.DictReader(file, fieldnames=("product_id", "user_id"))
+            records = [record for index, record in enumerate(csv_file) if index > 0]
+            counts[2] = len(records)
+            database["rentals"].insert_many(records)
+
+    return (tuple(counts), tuple(errors))
+
+def show_available_products(connect=None, database_name=None):
+    '''
+    Returns all products that are available to rent
+
+    :param connect: connection object, or None to use default
+    :param database_name: Database to use for the query,
+        or None to use production default
+    '''
+    connect = connect or MongoDBConnection()
+    database_name = database_name or config.DATABASE_NAME
+    with connect:
+        database = connect.connection.get_database(name=database_name)
+        products = database["products"]
+        result = dict()
+        for record in products.find({"quantity_available": {"$gt": 0}}):
+            result[record["product_id"]] = {
+                "description": record["description"],
+                "product_type": record["product_type"],
+                "quantity_available": record["quantity_available"]
```

Comment:
This dictionary seems very similar to the one you are given back from MongoDb. Are you doing it this way because you don't want to include something? You could make a copy, and delete the key you don't want, or do a comprehension:
```
resut[record["product_id"]] = {key: value for key, value in record.items() if key is not "one I don't want"}
```